src/core.js

```diff
@@ -1300,12 +1300,12 @@ $.extend( $.validator, {
 			this.settings.messages[ element.name ].remote = previous.message;
 
 			param = typeof param === "string" && { url: param } || param;
-
-			if ( previous.old === value ) {
+			optionDataString = JSON.stringify( $.param( $.extend( { data: value }, param.data ) ) );
```

Comment:
Is `$.param` really required? Wouldnt `stringify` already cover/support all we need?